internal/backend/kubernetes/backend.go

```diff
@@ -0,0 +1,178 @@
+package kubernetes
```

Comment:
This file is largely copied from /internal/hardware/kubernetes.go. The only noteworthy difference is the replacement of the `ByIP` method with `GetEC2Instance` given we need to satisfy the `/internal/frontend/ec2.Client` interface.